projects/ngx-cookie-service/src/lib/cookie.service.ts

```diff
@@ -14,7 +15,8 @@ export class CookieService {
   constructor(
     @Inject(DOCUMENT) private document: Document,
     // Get the `PLATFORM_ID` so we can check if we're in a browser.
-    @Inject(PLATFORM_ID) private platformId
+    @Inject(PLATFORM_ID) private platformId: any,
+    @Optional() @Inject(REQUEST_PROVIDER_TOKEN) private request: RequestProvider
```

Comment:
@valorkin I see you created `Injection Token` and supplied it here. Does it work for CSR along with SSR application? Would mind creating a sample repo that we can use to test this. 

@stevermeister take a look at this

Reply:
yes, now playing with it here - https://github.com/stevermeister/ssr-coookie